hicommonwealth/commonwealth
# Reuse fetched threads when switching b/w threads/topics

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/3599

## Description of Changes
- Now using already fetched data for threads, not making extra api calls
- Active threads from /activeThreads were unnecessarily (I assume) added to main thread store, removed that logic in [this commit](https://github.com/hicommonwealth/commonwealth/pull/3632/commits/449008adf32d216fc93a27e975ac7e4763d17dcd)
- Our effect to call /bulkThreads executed 2 times during component state update which caused the api call to run for the 2nd time which is unnecessary, added a timer to only call the api 1 time.

## Test Plan
- Go to any community
- Try switching b/w thread related pages
- Verify that the thread page only loads 1 time, when revisited without a page refresh it should not reload data from api
- Also verify that the /bulkThread api runs when the /discussions or any thread page is initially visited

## Deployment Plan
N/A

## Other Considerations
N/A
